webpack.hot.config.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * Webpack development server configuration
+ */
+
+'use strict';
+
+var webpack = require('webpack');
+var baseConfig = require('./webpack.config.js');
+
+module.exports = Object.assign({}, baseConfig, {
+  debug: true,
+
+  entry: [
+    'webpack/hot/only-dev-server'
+  ].concat(baseConfig.entry),
+
+  plugins: baseConfig.plugins.concat([
+    new webpack.HotModuleReplacementPlugin(),
+    new webpack.NoErrorsPlugin()
+  ])
+});
+
+module.exports.module.loaders[0].loader = 'react-hot!babel-loader';
```

Comment:
That wouldn't work because unshift returns the new array length, not the actual array. You'd have to use .concat for that.